docs/pipeline-api.md

```diff
@@ -2,7 +2,7 @@
 ---
 title: Pipeline API
 linkTitle: Pipeline API
-weight: 404
+weight: 1000
```

Comment:
Why this change here?

@geriom ^^^

Reply:
Definitely we should keep this as 404 (or anything other weight in the 400s).

Context: https://github.com/tektoncd/pipeline/pull/6117

Reply:
This is autogenerated, should I fix it manually?

Reply:
I think we may want to update the weight here
https://github.com/tektoncd/pipeline/blob/0b699455199cf1b80212ef1becade922883b0186/hack/update-reference-docs.sh#L26
@geriom